### PR TITLE
Topic recognition - ISICs UNet - 45576430

### PR DESCRIPTION
**Pull request by ChrisBailey**
Hello Shakes,

This request adds a new model for segmenting the ISICs 2018 skin lesion dataset using a U-Net Convolutional Network to the recognition module of PatternFlow. It solves problem 3 from the COMP3710 report and represents my submission of this assessment.

The code consists of two files, one (isicsunet.py) implementing a class for the ISICs U-Net model and a driver script (main.py) demonstrating use. Documentation is provided in the README and a figure providing sample output is also provided.

The code depends on Python3 (tested with 3.8), TensorFlow 2.x (tested with 2.3) and matplotlib (tested with 3.3)

A four layer padded U-Net is used, preserving skin features and mask resolution. The implementation utilises Adam as the optimizer and implements Dice distance as the loss function as this appeared to give quicker convergence than other methods (eg. binary cross-entropy).

The model was developed on a GTX 1660 TI (6GB VRAM) and certain values (notably batch size and image resolution) were set lower than might otherwise be ideal on more capable hardware. This is commented in the relevant code.

The utilised metric is a Dice coefficient implementation. My initial implementation appeared faulty and was replaced with a 3rd party implementation which appears correct. 3 epochs was observed to be generally sufficient to observe Dice coefficients of 0.8+ on test datasets but occasional non-convergence was observed and could be curbed by increasing the number of epochs. Visualisation of predictions is also implemented and shows reasonable correspondence. Orange bandaids represent an interesting challenge for the implementation as presented.

I hope you will find this a worthwhile addition to your library and look forward to your feedback.

Regards,
Christopher Bailey (45576430)

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
- Sufficiently detailed commit messages, a small number of them could be improved
- Code valid for Easy Difficulty
- Results appear okay

Other Feedback:
- I would not put the png file in the same director as the code
- Data visualisation and loading should not be part of the model class

**Comment by rexsw** [comment]
Code looks good for easy difficulty

For next time;

Dice lost and coefficient doesn’t have to be apart of a class they can just be their own function

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF NOT submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch)
